### PR TITLE
simple animals now have specific do_attack_animation visual effects.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
I have added and implemented on several simple mobs an `attack_vis_effect` variable and made `simple_animal/do_attack_animation()` use it. I also have done something similar to species datums and removed some so called shitcode. Last, I have sprited a new slash effect because since the define was already there.

## Why It's Good For The Game
Carps bite, hivebots claw, goats kick and space pirates slash; why must they have the same generic punch/smash attack visuals?

## Changelog
:cl:
fix: Uranium, bananium and bluespace golems and ethereals now properly show visual attack effects on do_attack_animation().
add: simple animals can now have their own visual attack effects instead of just punch or smash. e.g. carps bite and hivebots claw.
imageadd: the slash attack effect now has a sprite.
/:cl:

### CODE REVIEW

**Review by Twaticus — APPROVED** [review]

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1347,31 +1348,24 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
-					to_chat(user, "<span class='warning'>You can't bite with your mouth covered!</span>")
-					return FALSE
-				user.do_attack_animation(target, ATTACK_EFFECT_BITE)
-			else
-				user.do_attack_animation(target, ATTACK_EFFECT_PUNCH)
+			atk_verb = "kick"
+			atk_effect = ATTACK_EFFECT_KICK
+
+		if(atk_effect == ATTACK_EFFECT_BITE)
+			if(user.is_mouth_covered(FALSE, TRUE))
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species.dm:1357]: Give these named arguments.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
One minor nitpick.

**Comment by Ghommie** [comment]
```
FAIL: /datum/unit_test/food_edibility_check 0.6s
	REASON #1: [11:38:34] Runtime in icons.dm,607: Cannot read null.len
```
Yet Another Unrelated Unit Test Issue.

**Comment by Mothblocks** [comment]
Never seen that one before, weird.

**Pull request merged**